onnx_caffe2/bin/conversion.py

```diff
@@ -79,7 +79,8 @@ def caffe2_to_onnx(caffe2_net,
               help='Output path for the caffe2 init net file')
 def onnx_to_caffe2(onnx_model, output, init_net_output):
     onnx_model_proto = ModelProto()
-    onnx_model_proto.ParseFromString(onnx_model.read())
+    onnx_model_proto.ParseFromString(c2.optimize_onnx(onnx_model.read()))
```

Comment:
There are two ways of using onnx-caffe2, one is do file to file conversion (here), another is use the python interface to run an onnx model via caffe2 on the fly (which is in backend.py). Both will go through the same translation path `onnx_graph_to_caffe2_net`, so this `optimize_onnx` should be injected into there.

Reply:
that can probably be done, but it requires a more extensive refactoring because

- onnx_graph_to_caffe2_net operates on Graphs, and optimize-onnx needs the whole Model
- process() does destructive things before calling onnx_graph_to_caffe2_net, like stripping out the initializers